Marlin/src/module/motion.cpp

```diff
@@ -1468,23 +1468,23 @@ void set_axis_not_trusted(const AxisEnum axis) {
           phasePerUStep = 256 / (X_MICROSTEPS);
           phaseCurrent = stepperX.get_microstep_counter();
           effectorBackoutDir = -X_HOME_DIR;
-          stepperBackoutDir = INVERT_X_DIR ? -effectorBackoutDir : effectorBackoutDir;
+          stepperBackoutDir = INVERT_X_DIR ? effectorBackoutDir : -effectorBackoutDir;
           break;
       #endif
       #ifdef Y_MICROSTEPS
         case Y_AXIS:
           phasePerUStep = 256 / (Y_MICROSTEPS);
           phaseCurrent = stepperY.get_microstep_counter();
           effectorBackoutDir = -Y_HOME_DIR;
-          stepperBackoutDir = INVERT_Y_DIR ? -effectorBackoutDir : effectorBackoutDir;
+          stepperBackoutDir = INVERT_Y_DIR ? effectorBackoutDir : -effectorBackoutDir;
           break;
       #endif
       #ifdef Z_MICROSTEPS
         case Z_AXIS:
           phasePerUStep = 256 / (Z_MICROSTEPS);
           phaseCurrent = stepperZ.get_microstep_counter();
           effectorBackoutDir = -Z_HOME_DIR;
-          stepperBackoutDir = INVERT_Y_DIR ? -effectorBackoutDir : effectorBackoutDir;
+          stepperBackoutDir = INVERT_Z_DIR ? effectorBackoutDir : -effectorBackoutDir;
```

Comment:
I think sign ios not correct now.
Let's say invert=false, home dir = -1, phase =128 and stop pos =126
You need to move positive 2 ustep but on your computation you get `effectorBackoutDir=1` and `stepperBackoutDir=-1` then phaseDelta is -2 you will move 1022 positive

Reply:
The key on all this is the relationship between the marlin coordinate increasing on an non inverted stepper and the tmc phase increasing. They seem to be opposite to each other. I hope it is consistent across all TMC drivers. I'll read some data sheets now on the direction pin definition.

Reply:
From the 2209 datasheet:
> the microstep counter is increased (DIR=0) or decreased (DIR=1) with each STEP pulse

now we need to check what is the DIR state when marlin increases the axis coordinates, or maybe @thinkyhead knows from the top of his head.

Reply:
Yeah, I supposed invert=false microsteps increment but may not be true

Reply:
```c++
  #define SET_STEP_DIR(A)                       \
    if (motor_direction(_AXIS(A))) {            \
      A##_APPLY_DIR(INVERT_##A##_DIR, false);   \
      count_direction[_AXIS(A)] = -1;           \
    }                                           \
    else {                                      \
      A##_APPLY_DIR(!INVERT_##A##_DIR, false);  \
      count_direction[_AXIS(A)] = 1;            \
    }
```

with 
`#define X_APPLY_DIR(v,Q)  X_DIR_WRITE(v)`
`#define X_DIR_WRITE(STATE) WRITE(X_DIR_PIN,STATE)`

Then it seems negative count is with dir=0. My assumption was wrong

Reply:
I think it's better to add a comment on code, just to highlight this